### PR TITLE
feat: STOP-243 - create prism instance with full spec

### PR DESCRIPTION
**Pull request by brendarearden**
Addresses: STOP-243
**Summary**

Introduces `createAndCallPrismInstanceWithSpec` that allows you to pass in a prism config, prism request and an oas spec path or document and get a prism response or an error string back. 

**Checklist**

- The basics
  - [x] I tested these changes manually in my local or dev environment
- Tests
  - [x] Added or updated
  - [ ] N/A
- Event Tracking
  - [ ] I added event tracking and followed the event tracking guidelines
  - [x] N/A
- Error Reporting
  - [ ] I reported errors and followed the error reporting guidelines
  - [x] N/A

### CODE REVIEW

File: packages/http-server/src/__tests__/body-params-validation.spec.ts [hunk 1]
```diff
@@ -640,7 +640,7 @@ describe('body params validation', () => {
         });
 
         describe('and size bigger than 10MB', () => {
-          test('returns 422', async () => {
+          test('returns 413', async () => {
```
Comment (brendarearden) [inline packages/http-server/src/__tests__/body-params-validation.spec.ts:643]: this was checking a 413, so I updated the title to match. 

the rest of the changes in this file are formatting changes

File: packages/http/src/instanceWithSpec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          return response;
+        }, E.toError)
+      );
+    }),
+    TE.mapLeft((e: Error & { status?: number; additional?: { headers?: Dictionary<string> } }) => {
+      logger.error({ prismRequest }, `Unable to generate response: ${e}`);
+      return e.status || 500, JSON.stringify(ProblemJsonError.toProblemJson(e));
+    })
+  )();
+  if (isRight(result)) {
```
Comment (brendarearden) [inline packages/http/src/instanceWithSpec.ts:34]: trying to keep fp-ts from being needed outside of prism context by just returning the result directly.

File: packages/http/src/instanceWithSpec.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { Dictionary } from '@stoplight/types';
+
+export async function createAndCallPrismInstanceWithSpec(
+  document: string | object,
+  options: IHttpConfig,
+  prismRequest: IHttpRequest,
+  logger: pino.Logger
+) {
+  const operations = await getHttpOperationsFromSpec(document);
+  const prism = createInstance(options, { logger: logger.child({ name: 'PRISM INSTANCE' }) });
```
Comment (daniel-white) [inline packages/http/src/instanceWithSpec.ts:19]: can we have the client provide the logger and child we want? that way we dont have to export a pino type. we should have a logger contract in this library, not a 3rd party one

File: packages/http/src/instanceWithSpec.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    prism.request(prismRequest, operations),
+    TE.chainIOEitherK(response => {
+      return IOE.fromEither(
+        E.tryCatch(() => {
+          return response;
+        }, E.toError)
+      );
+    }),
+    TE.mapLeft((e: Error & { status?: number; additional?: { headers?: Dictionary<string> } }) => {
+      logger.error({ prismRequest }, `Unable to generate response: ${e}`);
```
Comment (daniel-white) [inline packages/http/src/instanceWithSpec.ts:30]: dont log out the whole request. it could have a huge payload and/or sensitive data

**Review by daniel-white — CHANGES_REQUESTED** [review]

File: packages/http/src/instanceWithSpec.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      return e.status || 500, JSON.stringify(ProblemJsonError.toProblemJson(e));
```
[the remark this answers is not in the document]
Comment (brendarearden) [reply packages/http/src/instanceWithSpec.ts:31]: this was definitely just a copy pasta leftover from the more server like attempt I did 😅 will be removing

File: packages/http/src/utils/operations.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+export async function getHttpOperationsFromSpec(specFilePathOrObject: string | object): Promise<IHttpOperation[]> {
+  const prismVersion = require('../../package.json').version;
+  const httpResolverOpts: HTTPResolverOptions = {
+    headers: {
+      'User-Agent': `PrismMockServer/${prismVersion} (${os.type()} ${os.arch()} ${os.release()})`,
+    },
+  };
+  const result = decycle(
+    await new $RefParser().dereference(specFilePathOrObject, { resolve: { http: httpResolverOpts } })
```
Comment (brendarearden) [inline packages/http/src/utils/operations.ts:22]: @daniel-white we are going to leave this here and then add a check for unresolved external references in our usage of this code and error at that point.

File: packages/http/src/instanceWithSpec.ts (hunk 3 above)
Comment (brendarearden) [reply packages/http/src/instanceWithSpec.ts]: its pino all the way down so this does need to remain pino at this time - I have opened an issue here https://github.com/stoplightio/prism/issues/2502 to address this later

**Review by EdVinyard — APPROVED** [review]

**Review by daniel-white — APPROVED** [review]

**Pull request merged**